lib/analyzer/image-inspector.ts

```diff
@@ -155,6 +171,22 @@ async function extractImageDetails(targetImage: string): Promise<ImageDetails> {
   };
 }
 
+function isLocalImageSameArchitecture(
+  platformOption: string,
+  inspectResultArchitecture: string,
+): boolean {
+  let platformArchitecture: string;
+  try {
+    // Note: this is using the same flag/input pattern as the new Docker buildx: eg. linux/arm64/v8
+    platformArchitecture = platformOption.split("/")[1];
+  } catch (error) {
+    debug(`Error parsing platform flag: '${error}'`);
+    return false;
+  }
+
+  return platformArchitecture === inspectResultArchitecture;
```

Comment:
this check is case sensitive, are we ok with? do we want to support users that write Arm insteadd of arm? (product question)

Reply:
Ah, great question, I'll ask Gareth.

Reply:
Basically, we should just support valid values and throw an error if wrong value provided, like docker does.The error handling will be added.